src/utils/fetchClient.ts

```
@@ -38,3 +43,34 @@ export const client = {
   patch: <T>(url: string, data: any) => request<T>(url, 'PATCH', data),
   delete: (url: string) => request(url, 'DELETE'),
 };
+
+export const getUsers = () => {
+  return client.get<User[]>('/users');
+};
+
+export const getPosts = ({ userId }: { userId: number }) => {
+  return client.get<Post[]>(`/posts?userId=${userId}`);
+};
+
+export const getComments = (postId: number) => {
+  return client.get<Comment[]>(`/comments?postId=${postId}`);
+};
+
+export const deleteComment = async (commentId: number) => {
+  try {
+    await client.delete(`/comments/${commentId}`);
+  } catch (error: any) {
+    console.error('Can not delete the comment', error.message);
+  } // finally {
+  //  return deleteComment;
+  // }
+};
```

Issue: The `deleteComment` function catches errors and only logs them, but does not re-throw or return a rejected promise. This prevents the calling code from knowing if the deletion failed, which can lead to incorrect UI updates. To meet the requirements, you should re-throw the error after logging it so the caller can handle the failure appropriately.